lib/msf/core/auxiliary/mikrotik.rb

```diff
@@ -0,0 +1,444 @@
+    # the format is a header line: `/ tree navigation`
+    # followed by commands: `set thing value`
+    def export_to_hash(config)
+      return {} unless config.is_a? String
+
+      config = config.gsub(/^\s{2,4}/, '') # replace code indents
+      config = config.gsub(/\\\s*\n/, '') # replace verbose multiline items as single lines, similar to terse
+      output = {}
+      header = ''
+      config.each_line do |line|
+        line = line.strip
+        # # jul/16/2020 14:26:57 by RouterOS 6.45.9
+        # typically the first line in the config
+        if %r{^# \w{3}/\d{2}/\d{4} \d{2}:\d{2}:\d{2} by (?<os>\w+) (?<version>[\d\.]+)$} =~ line
+          output['OS'] = ["#{os} #{version}"]
+
+        # terse format format is more 'cisco'-ish where header and setting is on one line
+        # /interface ovpn-client add connect-to=10.99.99.98 mac-address=FE:45:B0:31:4A:34 name=ovpn-out1 password=password user=user
+        # /interface ovpn-client add connect-to=10.99.99.98 mac-address=FE:45:B0:31:4A:34 name=ovpn-out2 password=password user=user
+        elsif %r{^(?<section>/[\w -]+)} =~ line && (line.include?(' add ') || line.include?(' set '))
+          [' add ', ' set '].each do |div|
+            next unless line.include?(div)
+
+            line = line.split(div)
+            if output[line[0].strip]
+              output[line[0].strip] << "#{div}#{line[1]}".strip
+              next
+            end
+            output[line[0].strip] = ["#{div}#{line[1]}".strip]
+          end
+
+        # /interface ovpn-client
+        # these are the section headers
+        elsif %r{^(?<section>/[\w -]+)$} =~ line
+          header = section.strip
+          output[header] = [] # initialize
+
+        # take any line that isn't commented out
+        elsif !line.starts_with?('#') && !header.empty?
+          output[header] << line.strip
+        end
+      end
+      output
+    end
+
+    # this takes a string of config like 'add connect-to=10.99.99.99 name=l2tp-hm password=123 user=l2tp-hm'
+    # and converts it to a hash of keys and values for easier processing
+    def values_to_hash(line)
+      return {} unless line.is_a? String
+
+      hash = {}
+      array = line.split(' ')
+      array.each do |setting|
+        key_value = setting.split('=')
+        unless key_value.length == 2
+          next # skip things like 'add'
+        end
+        # verbose mode gives empty fields, however our processing makes them "" instead of empty
+        # so we check if its empty and skip it to prevent a double quote or escaped double quote
+        # field from being loaded
+        next if key_value[1].strip == '""' || key_value[1].strip == '\\"\\"'
+
+        hash[key_value[0].strip] = key_value[1].strip
+      end
+      hash
+    end
+
+    def mikrotik_swos_config_eater(thost, tport, config)
+      credential_data = {
+        address: thost,
+        port: tport,
+        protocol: 'tcp',
+        workspace_id: myworkspace.id,
```

Comment:
If a user is not connected to the database (which happens frequently) this will fail because `workspace` will be nil, and they will get a stacktrace that complains `nil` has no method `id`
You could check if workspace is nil and simply not report it, or, like we did in pingback payloads, we can check if there is a database at all (https://github.com/rapid7/metasploit-framework/blob/2f9598e89a7428330e8fc01ec28b2d1a08d25064/lib/msf/core/payload/pingback.rb#L22)